pcmdi_metrics/precip_variability/scripts_pcmdi/calc_ps_area_freq_mean_regional.py

```diff
@@ -0,0 +1,152 @@
+import copy
+import datetime
+import json
+import os
+import pickle
+
+import numpy as np
+
+
+def prdday_to_frq3hridx(prdday, frequency):
+    frq3hr = 1.0 / (float(prdday) * 8.0)
+    idx = (np.abs(frequency - frq3hr)).argmin()
+    return int(idx)
+
+
+def prdday_to_frq1didx(prdday, frequency):
+    frq24hr = 1.0 / (float(prdday))
+    idx = (np.abs(frequency - frq24hr)).argmin()
+    return int(idx)
+
+
+# --------------------
+# User settings here
+# --------------------
+# "3hr" or "day"
+hr = ""
+# Input file name (pickle .pkl output from calc_ps_area.mean.py)
+fname = ""
+# Output directory (not including version)
+outdir = ""
+
+# -----------------------
+# The rest of the script
+# -----------------------
+ver = datetime.datetime.now().strftime("v%Y%m%d")
+
+if hr == "day":
+    frqs_forced = ["semi-annual", "annual"]
+    frqs_unforced = ["synoptic", "sub-seasonal", "seasonal-annual", "interannual"]
+elif hr == "3hr":
+    frqs_forced = ["semi-diurnal", "diurnal", "semi-annual", "annual"]
+    frqs_unforced = [
+        "sub-daily",
+        "synoptic",
+        "sub-seasonal",
+        "seasonal-annual",
+        "interannual",
+    ]
+
+infile = open(fname, "rb")
+psdm = pickle.load(infile)
+
+psdmfm = copy.deepcopy(psdm)
+for frc in psdm.keys():
+    if frc == "forced":
+        frqs = frqs_forced
+    elif frc == "unforced":
+        frqs = frqs_unforced
+    print(frc)
+    for mip in psdm[frc].keys():
+        print(mip)
+        for dat in psdm[frc][mip].keys():
+            frequency = np.array(psdm[frc][mip][dat]["freqs"])
+            del psdm[frc][mip][dat]["freqs"]
+            del psdmfm[frc][mip][dat]["freqs"]
+
+            for var in psdm[frc][mip][dat].keys():
+                print(dat, var)
+                for idm, dom in enumerate(psdm[frc][mip][dat][var].keys()):
+                    am = np.array(psdm[frc][mip][dat][var][dom])
+                    del psdmfm[frc][mip][dat][var][dom]
+                    psdmfm[frc][mip][dat][var][dom] = {}
+                    print(dom)
+                    for frq in frqs:
```

Comment:
@msahn In this loop, lines 74-130, I would like to double-check that all the time scales are being correctly sliced. If you could review this section I would greatly appreciate it.

Reply:
For semi-annual and annual, I use `np.amax` instead of `np.nanmean` to select maximum power within their range as below. This is to handle CMIP models that have different periods for semi-annual and annual cycles because they use different calendars (e.g., 360-day, 365-day, and Gregorian).

https://github.com/PCMDI/pcmdi_metrics/blob/035f654da98301538c8a1c693449435b346c37fb/pcmdi_metrics/precip_variability/lib/lib_variability_across_timescales.py#L433-L440

Reply:
@msahn Thank you for pointing this out! I have another question about the [sub-daily timescale indices](https://github.com/PCMDI/pcmdi_metrics/blob/035f654da98301538c8a1c693449435b346c37fb/pcmdi_metrics/precip_variability/lib/lib_variability_across_timescales.py#L443). If I'm interpreting this correctly, are we getting the average of data at frequencies equal to or larger than the sub-daily frequency?
```
amfm = np.nanmean(am[idx1 + 1 :])
```